activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -53,8 +53,8 @@ def initialize(connection, logger, connection_options, config)
 
         @statements = StatementPool.new(self.class.type_cast_config_to_integer(config[:statement_limit]))
 
-        if version < "5.1.10"
-          raise "Your version of MySQL (#{version_string}) is too old. Active Record supports MySQL >= 5.1.10."
+        if version < "5.7.9" || (mariadb? && version < "5.6.33")
```

Comment:
Required 5.7.9 is not for the first GA version of MySQL 5.7 but is for the first version of introduced `innodb_default_row_format` (`DYNAMIC` by default).

`innodb_default_row_format` is required for CI passed with utf8mb4, MariaDB 10.2.2 is the first version (and backported to 10.1.32), so if using MariaDB 10.1.31, probably CI would fail with this branch.

https://mariadb.com/kb/en/library/xtradbinnodb-server-system-variables/#innodb_default_row_format

Reply:
I did not know that. Updated to show different messages for MariaDB. It may not be necessary then please advise.